packages/core/test/reducers/core.test.ts

```diff
@@ -48,3 +50,199 @@ test('core reducer should support v7', t => {
+        schemaPath: '',
+        message: '',
+        params: {},
+        schema: undefined,
+        parentSchema: cloneDeep(schema.properties.foo)
+      },
+      {
+        keyword: '',
+        dataPath: 'foo',
+        schemaPath: '',
+        message: '',
+        params: {},
+        schema: undefined,
+        parentSchema: cloneDeep(schema.properties.foo)
+      },
+      {
+        keyword: '',
+        dataPath: 'foo.bar',
+        schemaPath: '',
+        message: '',
+        params: {},
+        schema: undefined,
+        parentSchema: cloneDeep(schema.properties.foo)
+      },
+    ]
+  };
+  const filtered = errorAt('foo', schema.properties.foo)(state);
+  t.is(filtered.length, 1);
+  t.deepEqual(filtered[0], state.errors[1]);
+});
+
+test('errorAt filters by schema', t => {
+  const schema: JsonSchema = {
+    type: 'object',
+    properties: {
+      foo: {
+        type: 'string',
+        const: 'bar'
+      }
+    }
+  };
+  const state: JsonFormsCore = {
+    data: undefined,
+    schema: undefined,
+    uischema: undefined,
+    errors: [
+      {
+        keyword: '',
+        dataPath: 'bar',
+        schemaPath: '',
+        message: '',
+        params: {},
+        schema: undefined,
+        parentSchema: cloneDeep(schema.properties.foo)
+      },
+      {
+        keyword: '',
+        dataPath: 'foo',
+        schemaPath: '',
+        message: '',
+        params: {},
+        schema: undefined,
+        parentSchema: cloneDeep(schema.properties.foo)
+      },
+      {
+        keyword: '',
+        dataPath: 'foo',
+        schemaPath: '',
+        message: '',
+        params: {},
+        schema: undefined,
+        parentSchema: {type: 'string', enum: ['bar']}
+      },
+    ]
+  };
+  const filtered = errorAt('foo', schema.properties.foo)(state);
+  t.is(filtered.length, 1);
+  t.deepEqual(filtered[0], state.errors[1]);
+});
+
+test('subErrorsAt filters by path', t => {
+  const schema: JsonSchema = {
+    type: 'object',
+    properties: {
+      foo: {
+        type: 'string',
+        const: 'bar'
+      }
+    }
+  };
+  const state: JsonFormsCore = {
+    data: undefined,
+    schema: undefined,
+    uischema: undefined,
+    errors: [
+      {
+        keyword: '',
+        dataPath: 'bar',
+        schemaPath: '',
+        message: '',
+        params: {},
+        schema: undefined,
+        parentSchema: cloneDeep(schema.properties.foo)
+      },
+      {
+        keyword: '',
+        dataPath: 'foo',
+        schemaPath: '',
+        message: '',
+        params: {},
+        schema: undefined,
+        parentSchema: cloneDeep(schema.properties.foo)
+      },
+      {
+        keyword: '',
+        dataPath: 'foo.bar',
+        schemaPath: '',
+        message: '',
+        params: {},
+        schema: undefined,
+        parentSchema: cloneDeep(schema.properties.foo)
+      },
+    ]
+  };
+  const filtered = subErrorsAt('foo', schema.properties.foo)(state);
+  t.is(filtered.length, 1);
```

Comment:
Maybe we should also add or reformulate a test where more than one item is matched as `subErrorsAt` is actually supposed to do that.